src/Umbraco.Web/PublishedCache/NuCache/DataSource/SerializerBase.cs

```diff
@@ -98,7 +98,9 @@ protected object ReadObject(char type, Stream stream)
                 case PrefixDateTime:
                     return PrimitiveSerializer.DateTime.ReadFrom(stream);
                 case PrefixByteArray:
-                    return PrimitiveSerializer.Bytes.ReadFrom(stream);
+                    // When it's a byte array always return as a LazyCompressedString
+                    // TODO: Else we need to make a different prefix for lazy vs eager loading
+                    return new LazyCompressedString(PrimitiveSerializer.Bytes.ReadFrom(stream));
```

Comment:
I think a different prefix must be used otherwise you can't serialize/deserialize a byte array that's not a string